pkg/spdk/disk.go

```diff
@@ -266,15 +268,23 @@ func addBlockDevice(spdkClient *SPDKClient, diskName, diskPath string, blockSize
 			return lvstore.UUID, nil
 		}
 
-		log.Infof("Renaming the existing lvstore %v to %v", lvstore.Name, lvstoreName)
-		renamed, err := spdkClient.BdevLvolRenameLvstore(lvstore.Name, lvstoreName)
-		if err != nil {
-			return "", errors.Wrapf(err, "failed to rename lvstore from %v to %v", lvstore.Name, lvstoreName)
-		}
-		if !renamed {
-			return "", fmt.Errorf("failed to rename lvstore from %v to %v", lvstore.Name, lvstoreName)
+		// Rename the existing lvstore to the name of the aio bdev
+		if diskUUID == lvstore.UUID {
+			log.Infof("Renaming the existing lvstore %v to %v", lvstore.Name, lvstoreName)
+			renamed, err := spdkClient.BdevLvolRenameLvstore(lvstore.Name, lvstoreName)
+			if err != nil {
+				return "", errors.Wrapf(err, "failed to rename lvstore from %v to %v", lvstore.Name, lvstoreName)
+			}
+			if !renamed {
+				return "", fmt.Errorf("failed to rename lvstore from %v to %v", lvstore.Name, lvstoreName)
+			}
+			return lvstore.UUID, nil
 		}
-		return lvstore.UUID, nil
+	}
+
+	// The lvstore should be created before, but it cannot be found now.
+	if diskUUID != "" {
+		return "", grpcstatus.Error(grpccodes.NotFound, fmt.Sprintf("cannot find lvstore with UUID %v", diskUUID))
 	}
 
 	return spdkClient.BdevLvolCreateLvstore(lvstoreName, diskName, defaultClusterSize)
```

Comment:
NIT: To make this function simpler, you can handle this (`diskUUID == ""`) first

Reply:
Updated. Thank you.